pkg/resources/credentials.go

```diff
@@ -282,6 +285,20 @@ func GetOpenstackCredentials(data CredentialsData) (OpenstackCredentials, error)
 	openstackCredentials := OpenstackCredentials{}
 	var err error
 
+	if spec.ApplicationCredentialID != "" {
+		openstackCredentials.ApplicationCredentialID = spec.ApplicationCredentialID
+		openstackCredentials.ApplicationCredentialSecret = spec.ApplicationCredentialSecret
+	} else if openstackCredentials.ApplicationCredentialID, _ = data.GetGlobalSecretKeySelectorValue(spec.CredentialsReference, OpenstackApplicationCredentialID); openstackCredentials.ApplicationCredentialID != "" {
+		openstackCredentials.ApplicationCredentialSecret, err = data.GetGlobalSecretKeySelectorValue(spec.CredentialsReference, OpenstackApplicationCredentialSecret)
+		if err != nil {
+			return OpenstackCredentials{}, err
+		}
+	}
+
+	if openstackCredentials.ApplicationCredentialID != "" {
+		return openstackCredentials, err
+	}
```

Comment:
```suggestion
	if spec.ApplicationCredentialID != "" {
		openstackCredentials.ApplicationCredentialID = spec.ApplicationCredentialID
		openstackCredentials.ApplicationCredentialSecret = spec.ApplicationCredentialSecret
		return openstackCredentials, err
	} else if openstackCredentials.ApplicationCredentialID, _ = data.GetGlobalSecretKeySelectorValue(spec.CredentialsReference, OpenstackApplicationCredentialID); openstackCredentials.ApplicationCredentialID != "" {
		openstackCredentials.ApplicationCredentialSecret, err = data.GetGlobalSecretKeySelectorValue(spec.CredentialsReference, OpenstackApplicationCredentialSecret)
		if err != nil {
			return OpenstackCredentials{}, err
		}
		return openstackCredentials, err
	}
```

This should get the same result and is a bit simpler